### PR TITLE
Feature: Markdown description support

### PR DESCRIPTION
**Pull request by DomHudson**
## Summary
Adding support for using markdown in package descriptions.

Enabled by existing functionality in [`readme_renderer`](https://github.com/pypa/readme_renderer/pull/55)

## Use
Simply add markdown to the existing the `long_description` field and set the `long_description_content_type` field to 'text/markdown'.

## Closes
https://github.com/devpi/devpi/issues/193

## Notes
The tox tests are failing on my local with py2.7 but passing if I run them manually with pytest and the same version of python. I believe this is because it is installing the unmodified devpi-server as part of the spinning up the 2.7 which doesn't contain the additional `long_description_content_type` field but may be wrong - feedback welcome.

I was a little confused by the stuff going on with `py.bultin._istext` and `py.bultin._totext` [here](https://github.com/devpi/devpi/blob/master/web/devpi_web/description.py#L27). In some cases, it seemed to encode the html as utf-8 and then decode it straight back again.

I've left it in as `devpi_web.description.DescriptionRenderer._force_text`.

### CODE REVIEW

File: server/devpi_server/model.py [hunk 1]
```diff
@@ -616,7 +616,7 @@ class PrivateStage(BaseStage):
         'metadata_version', 'summary', 'home_page', 'author', 'author_email',
         'maintainer', 'maintainer_email', 'license', 'description',
         'keywords', 'platform', 'classifiers', 'download_url',
-        'supported_platform', 'comment',
+        'description_content_type', 'supported_platform', 'comment',
```
Comment (fschulze) [inline server/devpi_server/model.py]: ``description_content_type`` is Metadata 2.1 according to https://github.com/pypa/twine/blob/0191a22a1ea51e4ff18fd20706094e03bb1abee4/twine/package.py#L145

I would add that and ``provides_extras`` as a separate commit.

I'll review the rest later.

Comment (fschulze) [reply server/devpi_server/model.py]: Also see 
https://www.python.org/dev/peps/pep-0566/

**Comment by DomHudson** [comment]
Okay thanks. I've added change logs and separated the metadata 2.1 (with `provides_extras`) via it's own section in `metadata_list_fields`.

Thanks
Dom

**Comment by fschulze** [comment]
I've rebase with master for an unrelated test failure fix and while at it squashed the commits into related ones.

**Comment by DomHudson** [comment]
Great thanks - let me know if you have other changes or if this good to go

Thanks,
Dom

**Comment by fschulze** [comment]
I removed the fixed time stuff in the tests, as they aren't needed apparently, maybe leftovers from development?

I also added a ``skipif`` marker to skip the test with older devpi versions, as we test with 4.2.1.

**Comment by fschulze** [comment]
Now waiting for the test results at https://travis-ci.org/fschulze/devpi/builds/443769088, as PRs aren't run with pre-release versions of the other devpi packages.

**Pull request merged**